src/Http/Headers/src/HeaderNames.cs

```diff
@@ -189,6 +189,9 @@ public static class HeaderNames
     /// <summary>Gets the <c>Pragma</c> HTTP header name.</summary>
     public static readonly string Pragma = "Pragma";
 
+    /// <summary>Gets the <c>Protocol</c> HTTP header name.</summary>
+    public static readonly string Protocol = ":protocol";
```

Comment:
Can you submit an [API proposal](https://github.com/dotnet/aspnetcore/issues/new?assignees=&labels=api-suggestion&template=30_api_proposal.md&title=) for this? I know there are going to be a bunch more APIs, but it might be good to learn the process by quickly filling out the issue template.

Reply:
I already did as part of my h2 WebSockets PR.

Reply:
API proposal: https://github.com/dotnet/aspnetcore/issues/42002